frontend/src/views/CellGuide/components/common/OntologyDagView/index.tsx

```diff
@@ -180,16 +195,62 @@ export default function OntologyDagView({
   // Animation duration - initially zero so the animation doesn't play on load
   const [duration, setDuration] = useState(0);
 
+  // construct explorer URL
+  const formattedSelectedOrganism = selectedOrganism
+    ?.toLowerCase()
+    ?.replace(/ /g, "_");
+  const formattedTissueId = tissueId.replace(/:/g, "_");
+  const formattedCellTypeId = cellTypeId?.replace(/:/g, "_");
+  let explorerUrl = "";
+
+  if (formattedTissueId && formattedCellTypeId && formattedSelectedOrganism) {
+    explorerUrl = `https://cellxgene.cziscience.com/e/cellguide-cxgs/tissues/${formattedSelectedOrganism}/${formattedTissueId}__${formattedCellTypeId}.cxg/`;
+  } else if (formattedCellTypeId && formattedSelectedOrganism) {
+    explorerUrl = `https://cellxgene.cziscience.com/e/cellguide-cxgs/${formattedSelectedOrganism}/${formattedCellTypeId}.cxg/`;
+  } else if (formattedTissueId && formattedSelectedOrganism) {
+    explorerUrl = `https://cellxgene.cziscience.com/e/cellguide-cxgs/tissues/${formattedSelectedOrganism}/${formattedTissueId}__CL_0000000.cxg/`;
+  }
+
+  const { data: validExplorerCxgs, isLoading: isLoadingValidExplorerCxgs } =
+    useValidExplorerCxgs();
+
+  const isExplorerCxgValid = useMemo(() => {
+    if (isLoadingValidExplorerCxgs || !validExplorerCxgs) return false;
+    const celltypeCxgs =
+      validExplorerCxgs.organism_celltype_cxgs[formattedSelectedOrganism];
+    const tissueCelltypeCxgs =
+      validExplorerCxgs.organism_tissue_celltype_cxgs[
+        formattedSelectedOrganism
+      ];
+    let valid = false;
+    if (tissueId && cellTypeId) {
+      valid = tissueCelltypeCxgs[tissueId]?.includes(cellTypeId);
+    } else if (cellTypeId) {
+      valid = celltypeCxgs?.includes(cellTypeId);
+    } else if (tissueId) {
+      valid = tissueCelltypeCxgs[tissueId]?.includes("CL:0000000");
+    }
+
+    return valid;
+  }, [
+    validExplorerCxgs,
+    isLoadingValidExplorerCxgs,
+    formattedSelectedOrganism,
+    tissueId,
+    cellTypeId,
+  ]);
+
   // The raw cell ontology tree data. This is called "rawTree" because it does not contain
   // the "isExpanded" property that is used to track the expanded state of each node, along with
   // other properties like the positions of the nodes.
-  const { data: rawTree } = useCellOntologyTree();
-
+  const { data: rawTree } = useCellOntologyTree(selectedOrganism);
```

Comment:
Ontology trees now must provide the organism name to get the corresponding tree and cell counts.